src/Acl/Policies/CorporationPolicy.php

```diff
@@ -96,6 +97,22 @@ private function isCeo(User $user, CorporationInfo $corporation)
         return in_array($corporation->ceo_id, $user->associatedCharacterIds());
     }
 
+    /**
+     * @param \Seat\Web\Models\User $user
+     * @param \Seat\Eveapi\Models\Character\CharacterInfo $character
+     * @return bool
+     */
+    private function isDirector(User $user, CorporationInfo $corporation)
+    {
+        $corporationRoles = CorporationRole::where('corporation_id', $corporation->corporation_id)
+            ->whereIn('character_id', $user->associatedCharacterIds())
+            ->where('role', 'Director')
+            ->where('type', 'roles')->get();
```

Comment:
why do you need to get here ? you just want to know if you have a match. You can use `exists`